src/metabase/db/metadata_queries.clj

```diff
@@ -27,12 +27,52 @@
       :data
       :rows))
 
+(defn- partition-field->filter-form
+  "Given a partition field, returns the default value can be used to query."
+  [field]
+  (let [field-form [:field (:id field) nil]]
+    (condp #(isa? %2 %1) (:base_type field)
+      :type/Number   [:> field-form -9223372036854775808]
+      :type/Date     [:> field-form "0001-01-01"]
+      :type/DateTime [:> field-form "0001-01-01 00:00:00"])))
```

Comment:
`:type/DateTime` is a superset of local datetimes and datetimes with offsets or zone IDs (`:type/DateTimeWithTZ`). This implementation might work for bigquery but is not guaranteed to work if there was a partition field with a `:type/DateTimeWithZoneOffset` base type in the future.

For bigquery the `_PARTITIONTIME` column will have a `:type/DateTimeWithLocalTZ` type, so this should work:
```suggestion
      :type/DateTimeWithLocalTZ [:> field-form "0001-01-01 00:00:00"])))
```

Reply:
@calherries clause for datetime is needed because we can partition on a datetime field. Added this test as an [example](https://github.com/metabase/metabase/pull/37102/commits/bbf83d66f180fabe6c83449effd92ba39ac750ae)

Reply:
Ah I stand corrected, in postgres and redshift it is fine to compare a timestamp with `:type/DateTimeWithZoneOffset` with a string literal that doesn't have any time zone information. This could be the case for other RDMBSs as well. And if it's not, a test will catch it. I'm okay with how it's implemented now then